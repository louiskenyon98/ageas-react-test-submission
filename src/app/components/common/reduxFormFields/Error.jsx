import React from 'react';
import PropTypes from 'prop-types';

class Error extends React.PureComponent {
    render() {
        if (this.props.message) {
            return (
                <div className="ui error message">
                    <div className="header">{this.props.message}</div>
                </div>
            )
        }
        return null
    }
}

Error.propTypes = {
    message: PropTypes.string
};

export default Error;
