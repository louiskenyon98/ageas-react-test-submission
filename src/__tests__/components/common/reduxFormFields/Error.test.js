import React from 'react';
import {shallow} from 'enzyme';
import Error from '../../../../app/components/common/reduxFormFields/Error';

describe('Error', () => {
    let props = {};
    let wrapper = () => shallow(<Error {...props}/>);
    beforeEach(() => {
        props = {
            message: 'error message'
        }
    });
    describe('render', () => {
        it('should render an error message when one is supplied', () => {
            expect(wrapper()).toMatchSnapshot();
        });
        it('should not render an error message when one is not supplied', () => {
            props.message = '';
            expect(wrapper()).toMatchSnapshot();
        })
    })
});
