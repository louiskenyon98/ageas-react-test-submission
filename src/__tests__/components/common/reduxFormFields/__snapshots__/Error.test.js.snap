// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error render should not render an error message when one is not supplied 1`] = `""`;

exports[`Error render should render an error message when one is supplied 1`] = `
<div
  className="ui error message"
>
  <div
    className="header"
  >
    error message
  </div>
</div>
`;
